src/containers/BooksForm.js

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import propTypes from 'prop-types';
+import { createBook } from '../actions/index';
+
+const categories = [
+  'Action',
+  'Biography',
+  'History',
+  'Horror',
+  'Kids',
+  'Learning',
+  'Sci-Fi',
+];
+
+class BooksForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: '',
+    };
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(event) {
+    const { name } = event.target;
+    const { state } = this;
+
+    this.setState({
+      ...state,
+      [name]: event.target.value,
+    });
```

Comment:
👍 , nice. Just take into account that state updates are automatically merged by `setState`, so no need to do it yourselves.

https://reactjs.org/docs/state-and-lifecycle.html#state-updates-are-merged